pkgs/development/python-modules/home-assistant-chip-wheels/default.nix

```diff
@@ -0,0 +1,305 @@
+  #
+  # yes this list of dependencies contains both build tools and proper dependencies.
+  env.PIP_NO_INDEX = "1";
+  env.PIP_FIND_LINKS =
+    let
+      dependencies = [
+        appdirs
+        appnope
+        black
+        build
+        colorama
+        coloredlogs
+        coverage
+        click
+        cryptography
+        diskcache
+        googleapis-common-protos
+        google-cloud-storage
+        ipython
+        jinja2
+        json5
+        jsonschema
+        lark
+        mobly
+        mypy
+        mypy-extensions
+        mypy-protobuf
+        packaging
+        parameterized
+        pip-tools
+        prompt-toolkit
+        protobuf
+        psutil
+        ptpython
+        pyelftools
+        pygments
+        pykwalify
+        pylint
+        pyperclip
+        pyserial
+        python-daemon
+        pyyaml
+        requests
+        setuptools
+        six
+        sphinx
+        sphinx-argparse
+        sphinx-design
+        stringcase
+        toml
+        tornado
+        types-protobuf
+        types-pyyaml
+        types-requests
+        types-setuptools
+        watchdog
+        websockets
+        wheel
+        yapf
+      ];
+      depListToAttrs =
+        list:
+        builtins.listToAttrs (
+          map (dep: {
+            name = dep.name;
+            value = dep;
+          }) list
+        );
+      saturateDependencies =
+        deps:
+        let
+          before = deps;
+          new = lib.mergeAttrsList (
+            map (dep: depListToAttrs (dep.propagatedBuildInputs or [ ])) (lib.attrValues before)
+          );
+          after = before // new;
+        in
+        if lib.attrNames before != lib.attrNames after then saturateDependencies after else before;
+      saturateDependencyList = list: lib.attrValues (saturateDependencies (depListToAttrs list));
+      saturatedDependencyList = lib.filter (drv: drv ? dist) (saturateDependencyList dependencies);
+    in
+    lib.concatMapStringsSep " " (dep: "file://${dep.dist}") saturatedDependencyList;
+
+  configurePhase =
+    let
+      configureArguments = [
+        ''chip_project_config_include_dirs=["//.."]''
+        ''chip_crypto="boringssl"''
+        ''enable_rtti=true''
+        ''chip_config_memory_debug_checks=false''
+        ''chip_config_memory_debug_dmalloc=false''
+        ''chip_mdns="minimal"''
+        ''chip_minmdns_default_policy="libnl"''
+        ''chip_python_version="${lib.versions.majorMinor python3.version}"''
+        ''chip_python_platform_tag="any"''
+        ''chip_python_package_prefix="home-assistant-chip"''
+        ''custom_toolchain="custom"''
+        ''target_cc="${stdenv.cc.targetPrefix}cc"''
+        ''target_cxx="${stdenv.cc.targetPrefix}c++"''
+        ''target_ar="${stdenv.cc.targetPrefix}ar"''
+      ];
+    in
+    ''
+      runHook preConfigure
+
+      gn --root=. gen --check --fail-on-unused-args out ${
```

Comment:
tiny optional nit: nixpkgs already has gn build hooks (https://github.com/NixOS/nixpkgs/blob/6ced9c6711a42e6b1195384709c4af2f79ea7893/pkgs/development/tools/build-managers/gn/setup-hook.sh#L11), so you could use the standard buildPhase and just set `gnFlags` to what is in `configureArguments`